src/notation/internal/notationinteraction.cpp

```diff
@@ -1859,14 +1861,21 @@ bool NotationInteraction::applyPaletteElement(mu::engraving::EngravingItem* elem
         } else if (element->isSLine() && element->type() != ElementType::GLISSANDO) {
             mu::engraving::Segment* startSegment = sel.startSegment();
             mu::engraving::Segment* endSegment = sel.endSegment();
+
+            ByteArray a = element->mimeData();
+            mu::engraving::XmlReader e(a);
+            mu::engraving::Fraction duration;        // dummy
+            PointF dragOffset;
+            mu::engraving::ElementType type = mu::engraving::EngravingItem::readType(e, &dragOffset, &duration);
+
             bool firstStaffOnly = isSystemTextLine(element) && !(modifiers & Qt::ControlModifier);
             staff_idx_t startStaff = firstStaffOnly ? 0 : sel.staffStart();
             staff_idx_t endStaff   = firstStaffOnly ? 1 : sel.staffEnd();
             for (staff_idx_t i = startStaff; i < endStaff; ++i) {
-                mu::engraving::Spanner* spanner = static_cast<mu::engraving::Spanner*>(element->clone());
-                spanner->setScore(score);
-                spanner->styleChanged();
-                score->cmdAddSpanner(spanner, i, startSegment, endSegment, modifiers & Qt::ControlModifier);
+                mu::engraving::Spanner* s = static_cast<mu::engraving::Spanner*>(engraving::Factory::createItem(type, score->dummy()));
+                rw::RWRegister::reader()->readItem(s, e);
```

Comment:
Except for consistency, does this change have a specific effect? If not, I'm not sure how nice this is, because going via mime data is definitely less efficient than just cloning directly

Reply:
Partly consistency, partly the fact that cloning doesn't seem to produce a text line with "system" style (e.g. font size). Unsure why this is the case, I'll look into it.

Reply:
Alright @cbjeukendrup - just pushed a different approach using clones & fixed the unit tests.